### PR TITLE
168 temporal validation

### PR DESCRIPTION
**Pull request by remeryAGS**
Start dates and latency for #168

### CODE REVIEW

File: gips/data/landsat/landsat.py [hunk 1]
```diff
@@ -297,39 +303,53 @@ class landsatData(Data):
...
+            'toa': True,
+            'startdate': date(1984, 3, 1),
+            'latency': 1,
         },
         'bqa': {
             'assets': ['DN'],
             'description': 'LC8 band quality',
-            'toa': True
+            'toa': True,
+            'startdate': date(1984, 3, 1),
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: This is a Landsat 8 only product and should have a start date of `date(2013, 5, 30)`.

File: gips/data/landsat/landsat.py [hunk 2]
```diff
@@ -297,39 +303,53 @@ class landsatData(Data):
...
         'tcap': {
             'assets': ['DN'],
             'description': 'Tassled cap transformation',
-            'toa': True
+            'toa': True,
+            'startdate': date(1984, 3, 1),
+            'latency': 1,
         },
-        'dn': {
+        'dt': {
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: This is an erroneous change.  `dn` is for digital number.

File: gips/data/landsat/landsat.py [hunk 3]
```diff
@@ -340,78 +360,108 @@ class landsatData(Data):
                 'Z: cloud height in meters (default: 4000)'
             ],
             'nargs': '*',
-            'toa': True
+            'toa': True,
+            'startdate': date(1984, 3, 1),
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: also Landsat 8 only.  2013-05-30.

File: gips/data/landsat/landsat.py [hunk 4]
```diff
@@ -340,78 +360,108 @@ class landsatData(Data):
...
-            'arguments': [__toastring]
+            'arguments': [__toastring],
+            'startdate': date(1984, 3, 1),
+            'latency': 1,
         },
         # NEW!!!
         'ndvi8sr': {
             'assets': ['SR'],
             'description': 'Normalized Difference Vegetation from LC8SR',
+            'startdate': date(1984, 3, 1),
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: Landsat 8 only.

File: gips/data/landsat/landsat.py [hunk 5]
```diff
@@ -340,78 +360,108 @@ class landsatData(Data):
...
+            'latency': 1,
         },
         'landmask': {
             'assets': ['SR'],
             'description': 'Land mask from LC8SR',
+            'startdate': date(1984, 3, 1),
+            'latency': 1,
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: Landsat 8 only (2013-5-30)

File: gips/utils.py [hunk 6]
```diff
@@ -269,22 +269,30 @@ def get_data_variables():
     # ex asset = modis
     for driver in sources.keys():
         data_class = import_data_class(driver)
+        print(data_class)
```
Comment (ags-tolson) [inline gips/utils.py]: optional:  In general we're trying to push gips towards acting like a library by default, so we try to avoid print() for new code.

Comment (remeryAGS) [inline gips/utils.py]: Whoops, I always forget at least 1 print statement

File: gips/inventory/dbinv/models.py [hunk 7]
```diff
@@ -260,4 +262,4 @@ class PostProcessJobs(models.Model):
 
     class Meta:
         unique_together = ('job', 'args')
-        
+
```
Comment (ags-tolson) [inline gips/inventory/dbinv/models.py]: Thanks for killing all this whitespace!

**Comment by ags-tolson** [comment]
ok with me if this merges, pending ircwaves' approval of the third commit

File: gips/data/landsat/landsat.py [hunk 8]
```diff
@@ -276,17 +279,23 @@ class landsatData(Data):
         'rad': {
             'assets': ['DN'],
             'description': 'Surface-leaving radiance',
-            'arguments': [__toastring]
+            'arguments': [__toastring],
+            'startdate': self._lt5_startdate,
```
Comment (ircwaves) [inline gips/data/landsat/landsat.py]: there's no `self` at this scope.  just a naked `_lt5_startdate` (or `_lc8_startdate` as the case may be).

Comment (ircwaves) [inline gips/data/landsat/landsat.py]: applies throught this `dict`

Comment (remeryAGS) [inline gips/data/landsat/landsat.py]: Arg too much time in javascript land!

Comment (ircwaves) [inline gips/data/landsat/landsat.py]: Minimum dev test:
```
python -c 'from gips.data.landsat import landsatData'
```
;)

Comment (justinfisk) [inline gips/data/landsat/landsat.py]: extreme minimum

Comment (remeryAGS) [inline gips/data/landsat/landsat.py]: Dev suggestion: test on correct branch

**Pull request merged**